worker/backup_processor.go

```diff
@@ -65,7 +60,7 @@ func NewBackupProcessor(db *badger.DB, req *pb.BackupRequest) *BackupProcessor {
 	bp := &BackupProcessor{
 		DB:      db,
 		Request: req,
-		threads: make([]*threadLocal, backupNumGo),
+		threads: make([]*threadLocal, x.WorkerConfig.Badger.GetUint64("goroutines")),
```

Comment:
TODO: Figure out how to calculate this at runtime so we never have OOM errors with default flags?